components/job-orchestration/job_orchestration/search_scheduler/search_scheduler.py

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+import time
+from pathlib import Path
+from typing import Dict, List, Optional
+
+import mysql.connector
+
+import msgpack
+import pathlib
+
+from celery import group
+
+from clp_py_utils.clp_config import SEARCH_JOBS_TABLE_NAME, CLP_METADATA_TABLE_PREFIX
+from job_orchestration.executor.search.fs_search_task import search
+from job_orchestration.job_config import SearchConfig
+
+from pydantic import ValidationError
+
+from clp_py_utils.clp_config import CLPConfig
+from clp_py_utils.clp_logging import get_logger, get_logging_formatter, set_logging_level
+from clp_py_utils.core import read_yaml_config_file
+
+from .common import JobStatus  # type: ignore
+
+# Setup logging
+logger = get_logger("search-job-handler")
+
+class SearchJob:
+    def __init__(self, async_task: any) -> None:
+        self.async_task: any = async_task
+
+# dictionary of active jobs indexed by job id
+active_jobs : Dict[str, SearchJob] = {}
+
+def cancel_job(job_id):
+    global active_jobs
+    active_jobs[job_id].async_task.revoke(terminate=True)
+    try:
+        active_jobs[job_id].async_task.get()
+    except:
```

Comment:
```suggestion
    except Exception:
```

Otherwise, this could catch exceptions like `KeyboardException` (CTRL-c).